src/plugins/navsat_map/NavSatMap.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "NavSatMap.hh"
+
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include <ignition/common/Console.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+
+#include "ignition/gui/Application.hh"
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class NavSatMapPrivate
+  {
+    /// \brief List of topics publishing navSat messages.
+    public: QStringList topicList;
+
+    /// \brief Holds data to set as the next navSat
+    public: msgs::NavSat navSatMsg;
+
+    /// \brief Node for communication.
+    public: transport::Node node;
+
+    /// \brief Mutex for accessing navSat data
+    public: std::recursive_mutex navSatMutex;
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+NavSatMap::NavSatMap()
+  : Plugin(), dataPtr(new NavSatMapPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+NavSatMap::~NavSatMap()
+{
+}
+
+/////////////////////////////////////////////////
+void NavSatMap::LoadConfig(const tinyxml2::XMLElement *_pluginElem)
+{
+  // Default name in case user didn't define one
+  if (this->title.empty())
+    this->title = "Message display";
```

Comment:
Should the title be `"NavSatMap display"` instead?

Reply:
Ah nice catch! I changed it to "Navigation satellite map": c898d07